design/accepted/gc_plus/gc-plus-milestone2-execution-plan.md

```diff
@@ -0,0 +1,53 @@
+# Uncommitted Garbage Collection - Milestone 2 Execution Plan
+
+Uncommitted Garbage Collection [Proposal](https://github.com/treeverse/lakeFS/blob/master/design/accepted/gc_plus/uncommitted-gc.md)
+
+## Milestone 1
+
+The first beta version that was released included:
+1. Implementation of the clean run flow for old and new repository structures (without optimizations)
+2. Mark & Sweep
+3. Integration tests
+4. Backup & Restore - minimal support using rclone
+
+## Milestone 2
+
+### Goals
+1. Removing the limitation of a read-only lakeFS during the GC+ job run
+2. Performance improvements - better parallelization of the storage namespace traversal
+3. Implementing Optimized (Incremental) Run
+
+### Non-Goals
+1. Support for non-S3 repositories
+   * Azure
+   * GCP
+2. Incorporation of committed & uncommitted GC into a single job
+    * Including GC changes, configuration, and behavior changes to fit GC
+3. Metrics and Logging additions
+4. Deployment to lakeFS Cloud
+5. Improve Backup & Restore
+
+### Plan
```

Comment:
Thanks for listing all the dependencies and the tasks themselves! It really made the difference for me and helped me understand what we're facing.